### ISSUE TITLE
Design - Zoom controls

### ISSUE THREAD

**Issue by thisandagain** [LLK/scratch-blocks#43]
In Scratch on web / desktop we have zoom controls in the bottom right of the editor:
![image](https://cloud.githubusercontent.com/assets/747641/12758819/22f57664-c9ac-11e5-82d5-cb33f461a449.png)

For tablet and desktop / web, we should make a decision as to what we want:
- No controls (no zooming)
- Button controls
- Pinch to zoom controls

One consideration is that "pinch to zoom" will impact the potential for multitouch interaction with blocks.

/cc @tmickel

**Comment by tmickel** [comment]
@carljbowman This is on the list for the prototype, so we'll need a design.

**Comment by carljbowman** [comment]
Assigning both @rachel-fenichel and @picklesrus – Was not sure who to assign as I know there are refactors going on in Blockly that may impact these changes. Feel free to reassign.

_We held off on exploring pinch zoom in touch environments, namely the I/O prototype. As we focus on the desktop experience, pinch zoom will remain out of scope._

### Zoom In/Out Behavior
Overall we should keep Blockly's zoom behavior: zooming towards the center of the workspace viewport. Since the viewport includes the Block Palette and in Scratch Blocks we always have the palette open the zoom feels a bit off. We need to account for the blocks palette width (vertical grammar) or height (horizontal grammar) when zooming.

<img width="557" alt="screen shot 2017-01-17 at 9 14 04 am" src="https://cloud.githubusercontent.com/assets/3409578/22023736/63e4f184-dc95-11e6-8dde-2bafe6586fdd.png">

### Reset Zoom Control Behavior
Currently Blockly does two things when resetting: returns to 100% zoom and repositions the viewport to the center of the blocks. This centering behavior can be a bit disorienting and should be removed. 

Reset should instead just return to 100% zoom and position to the center of the viewport, similarly to how zooming in/out positions.

### Zoom Controls Postion
If this is not already implemented within the code, we should allow the controls to be positionable in the corners of the workspace: top-left, top-right, bottom-left, bottom-right. Default position in Scratch-blocks should be bottom-right (or in RTL bottom-left). 

### Design & Icon Assets
<img width="547" alt="screen shot 2017-01-17 at 9 35 10 am" src="https://cloud.githubusercontent.com/assets/3409578/22024539/499d6b96-dc98-11e6-85e3-50059bf93ce7.png">

[zoom-icons.zip](https://github.com/LLK/scratch-blocks/files/711319/zoom-icons.zip)

### Scroll Gestures
The current behavior in [scratch-gui](https://llk.github.io/scratch-gui/) feels appropriate: scrolling should move (scroll) the workspace in rather than zooming the workspace, which is the current behavior within the [Blockly Playground](https://blockly-demo.appspot.com/static/tests/playground.html).

**Comment by rachel-fenichel** [comment]
More spec, from an offline discussion with @carljbowman:
The zoom buttons/icons on a layer above the blocks by default. When dragging the workspace the blocks move above the icons.  When dragging blocks/block stacks, the dragging blocks move above the icons.
The icons should be opaque with a button around them.

**Comment by thisandagain** [comment]
@carljbowman From the assets here (particularly due to the mock being zoomed in) there are still a few questions:
- What are the outer dimensions of these buttons (the assets you provided are ~10x10px / existing zoom controls are 32x32px)?
- What is the vertical margin between them?
- Is there a hover state?

Alternatively: having the latest general GUI mockup easily available in vector could help alleviate these questions as well.

**Comment by thisandagain** [comment]
Quick first pass:

Current status at 300% zoom (compare with above both at 100%):
![image](https://cloud.githubusercontent.com/assets/747641/23097068/a0e6efd4-f5f8-11e6-81e3-63435e9a2d93.png)

Current status at 100% zoom:
![image](https://cloud.githubusercontent.com/assets/747641/23097081/ddf0a1b8-f5f8-11e6-97e3-8bfa1d9cccb7.png)

I switched these over to SVG (currently renders from `sprites.png`), but SVG rendering thins out the strokes (differences in anti aliasing). Looks like they are breaking up / hard to parse at 100% zoom. @carljbowman and @lifeinchords any thoughts? My instinct is to bulk up the sizing and strokes a bit.

WIP commit here: https://github.com/thisandagain/scratch-blocks/commit/fd95bb194aa3f823b3b97dc227a1bdcfe5e3bc7f

**Comment by cpseager** [comment]
Would putting 100% as the middle icon (as it was in 2) make its function more obvious?
